src/AWS.Deploy.Orchestration/RecipeHandler.cs

```diff
@@ -225,5 +234,31 @@ private bool ContainsRecipeFile(string directoryPath)
 
             return recipeFilePaths.All(filePath => Path.GetFileNameWithoutExtension(filePath).Equals(directoryName, StringComparison.Ordinal));
         }
+        
+        /// Creates an option setting item dependency tree that indicates
+        /// which option setting items need to be validated if a value update occurs.
+        /// The function recursively goes through all the settings and their children to build this tree.
+        /// This method also creates a Leaf Id which will help reference <see cref="OptionSettingItem"/>.
+        /// </summary>
+        private void BuildDependencyTree(RecipeDefinition recipe, List<OptionSettingItem> optionSettingItems, Dictionary<string, List<string>> dependencyTree, string parentLeafId = "")
+        {
+            foreach (var optionSettingItem in optionSettingItems)
+            {
+                optionSettingItem.LeafId = $"{parentLeafId}{optionSettingItem.Id}";
+                optionSettingItem.ParentId = parentLeafId.EndsWith(".") ? parentLeafId.Substring(0, parentLeafId.Length - 1) : parentLeafId;
+                foreach (var dependency in optionSettingItem.DependsOn)
+                {
+                    if (dependencyTree.ContainsKey(dependency.Id))
```

Comment:
Shouldn't the key in this dictionary be the settings full id? Meaning if a recipe had 2 `CreateNew` setting ids under 2 different object settings those ids would get grouped together here.

Reply:
https://github.com/aws/aws-dotnet-deploy/blob/aacb712b5be4f1816ad9cb6933927c1b4d93811c/src/AWS.Deploy.Common/Recipes/PropertyDependency.cs#L11